osl/interface_linux.go

```diff
@@ -307,14 +307,26 @@ func (n *networkNamespace) AddInterface(srcName, dstPrefix string, options ...If
 
 func configureInterface(iface netlink.Link, i *nwIface) error {
 	ifaceName := iface.Attrs().Name
+	address := func(i *nwIface) string {
+		if i.Address() == nil {
+			return "(no address)"
+		}
+		return fmt.Sprintf("%q", i.Address())
+	}
+	address6 := func(i *nwIface) string {
+		if i.AddressIPv6() == nil {
+			return "(no address)"
+		}
+		return fmt.Sprintf("%q", i.AddressIPv6())
+	}
 	ifaceConfigurators := []struct {
 		Fn         func(netlink.Link, *nwIface) error
 		ErrMessage string
 	}{
 		{setInterfaceName, fmt.Sprintf("error renaming interface %q to %q", ifaceName, i.DstName())},
 		{setInterfaceMAC, fmt.Sprintf("error setting interface %q MAC to %q", ifaceName, i.MacAddress())},
-		{setInterfaceIP, fmt.Sprintf("error setting interface %q IP to %q", ifaceName, i.Address())},
```

Comment:
Hmm, doing that instead doesn't prevent the segfault.
I looked a little closer, and I was incorrect about the effect of this segfault: the runtime catches it (even in the original version), notices that the segfault happened while calling Error() or String() on a nil pointer, ignores the segfault because it special-cases it, and produces "<nil>" in the output message.  I'm attached with a debugger, so I can continue past these when they crop up, but I wouldn't have guessed that it was supposed to work like that.
Long story short, we can either patch to avoid passing the pointer without checking it first, or we can let the runtime keep doing its thing and close this without merging.